LibOS/shim/src/bookkeep/shim_vma.c

```diff
@@ -209,15 +210,32 @@ static void * current_heap_top;
 
 int init_vma (void)
 {
+    for (int i = 0 ; i < RESERVED_VMAS ; i++)
+        reserved_vmas[i] = &early_vmas[i];
+
     if (!(vma_mgr = create_mem_mgr(init_align_up(VMA_MGR_ALLOC)))) {
         debug("failed allocating VMAs\n");
         return -ENOMEM;
     }
 
-    create_lock(vma_list_lock);
+    for (int i = 0 ; i < RESERVED_VMAS ; i++) {
+        struct shim_vma * new = get_mem_obj_from_mgr(vma_mgr);
+        assert(new);
```

Comment:
Why not return -ENOMEM here? This can happen in runtime, `assert` is for things which are intended to be impossible to reach.